src/iterate_matrix.F

```diff
@@ -982,6 +1083,268 @@ SUBROUTINE matrix_sqrt_Newton_Schulz(matrix_sqrt, matrix_sqrt_inv, matrix, thres
       CALL timestop(handle)
 
    END SUBROUTINE matrix_sqrt_Newton_Schulz
+! **************************************************************************************************
+!> \brief ...
+!> \param n ...
+!> \return ...
+! **************************************************************************************************
+   FUNCTION factorial(n) RESULT(res)
```

Comment:
This one-liner function is used only three times, hence inlining would be shorter.